bbc/simorgh
# WSTEAM1-423 NextJS Cypress E2E Using Existing Setup

Resolves JIRA [WSTEAM1-423](https://jira.dev.bbc.co.uk/browse/WSTEAM1-423)

Overall changes
======
Enables our existing Cypress setup to accommodate the NextJS app.

There's a slight issue where importing in TS files causes the cypress to break, and I couldn't figure out the necessary TS configuration needed to get it to work. So any imports were removed that were TS files.

Code changes
======

- Adds Cypress and the various packages as dev dependencies within the NextJS app
- Mirrors the scripts from Simorgh for E2E's for the NextJS app
- Copy and append to a cypress.json specifically for NextJS pointing to the OG directories
- Ignores live page tests within the Simorgh cypress json config.
- Adds GitHub workflow step for NextJS E2E running
- Adds functionality to `getPageData` custom cypress function to work for NextJS
- Adds example test using @karinathomasbbc 's pattern for Home Pages, that can only be ran on test due to existing BFF cert issues for CI

Testing
======
1. Run Cypress
2. Check tests are running, there is a gated check just for test, so you'll need to adjust that to get it run locally.

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
